src/components/PostDetails.tsx

```diff
@@ -5,31 +5,45 @@ import { NewCommentForm } from './NewCommentForm';
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+
+import {
+  fetchComments,
+  setError as setCommentsError,
+  setLoading as setCommentsLoading,
+  addComment as addCommentToState,
+  deleteComment as removeCommentFromState,
+} from '../features/comments/commentsSlice';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const {
+    items: comments,
+    loaded: commentsLoaded,
+    hasError: commentsError,
+  } = useAppSelector(state => state.comments);
+
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(setCommentsLoading(false));
+    dispatch(setCommentsError(false));
     setVisible(false);
 
     commentsApi
       .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .then(commentsList => dispatch(fetchComments(commentsList))) // save the loaded comments
```

Comment:
By the way, you could use createAsyncThunk to handle all promise states in reducer instead of dispatching a few different actions here

But all previous comments solved - approved ✔️